### PR TITLE
feat: adds playbook and role to download and setup aws_logs_agent

### PR DESCRIPTION
**Pull request by DubeySandeep**
## Description

This PR adds a playbook for cloudwatch aws logs agent setup which will be further used for logging server logs to cloudwatch.

**Ticket link:** https://tasks.opencraft.com/browse/BB-6085

## Testing instructions
Prereq: All the aws resources are created for storing server logs on cloudwatch ([steps](https://github.com/open-craft/terraform-scripts/pull/40))
1. Add similar variable in any config file
```
CLOUDWATCH_LOG_FILES:
  - filename: /edx/var/log/lms/edx.log
    group_name: uq-staging-elasticsearch-cloudwatch-log-group
    stream_name: uq-staging-elasticsearch-cloudwatch-log-stream
  - filename: /edx/var/log/cms/edx.log
    group_name: uq-staging-elasticsearch-cloudwatch-log-group
    stream_name: uq-staging-elasticsearch-cloudwatch-log-stream

AWS_EC2_INSTANCE_REGION: ap-southeast-2
```
2. Run the following command to test the playbook
`ansible-playbook --extra-vars=@<filepath-to-config-variables> -i <inventory-path> -l <hosts-name> playbooks/openedx_native.yml `

Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A SRE team member has approved the PR if it is code shared across multiple services and you don't own all of the services.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a SRE ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/spaces/SRE/pages/28967861/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).
  - [ ] Think about how this change will affect Open edX operators.  Have you updated the wiki page for the next Open edX release?

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @DubeySandeep! I've created [OSPR-6744](https://openedx.atlassian.net/browse/OSPR-6744) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by natabene** [comment]
@DubeySandeep Thank you for your contribution. Is this ready for our review?

File: playbooks/deploy_nginx_for_dbt_docs.yml [hunk 1]
```diff
@@ -5,5 +5,7 @@
   roles:
     - aws
     - nginx
-    - aws_cloudwatch_agent
-    - dbt_docs_nginx
\ No newline at end of file
+    - dbt_docs_nginx
+    - role: aws_cloudwatch_agent
+      vars:
+          template_src: cloudwatch-agent-metrics-config.json
```
Comment (0x29a) [inline playbooks/deploy_nginx_for_dbt_docs.yml:11]: With this, we're overriding the default `amazon-cloudwatch-agent.json` without possibility to set it back. Instead, we should turn `amazon-cloudwatch-agent.json` into a template, and extend it with our logging configuration.

**Review by 0x29a — CHANGES_REQUESTED** [review]

**Comment by 0x29a** [comment]
@natabene, this is under our internal review. I'll let you know once it's ready.

**Comment by 0x29a** [comment]
@natabene, we're going to continue the work on this in another pull request. Could you please close the current one?

**Comment by openedx-webhooks** [comment]
@DubeySandeep Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/configuration&entry.867055334=https://github.com/openedx/configuration/pull/6767&entry.1484655318=https://github.com/DubeySandeep&entry.752974735=2022-05-30+12:29&entry.1917517419=2022-06-21+17:45&entry.2133058324=No) so we can improve your experience in the future.

**Pull request closed without merging**